ISICbalance/ViewModels/AccountViewModel.swift

```diff
@@ -13,13 +13,16 @@ import SwiftKeychainWrapper
 
 class AccountViewModel: BaseViewModel {
     let keychainManager: KeychainManager
-    
     let username = MutableProperty<String>("")
     let password = MutableProperty<String>("")
     var validationSignal: Property<Bool>
     var validationErrors: Property<[LoginValidation]>
 
-    lazy var loginAction = Action<(),(),LoginError> { [unowned self] in
+    lazy var loginAction = Action<(),(),LoginError> { [weak self] in
```

Comment:
pro tip: pokud to jde, snaž se všechno dělat `let`, tuhle akci jsi mohl nastavit v `init`u

Reply:
Moc nevím, jak to mám poskládat. Pokud tu akci nastavuji po zavolání super.init(), tak mi xcode erroruje "Property 'self.loginAction' not initialized at super.init call" a pokud to zavolám před super.init(), tak to zase nezná self uvnitř akce.

Reply:
U BalanceVM jsem to vyřešil tak, že RequestManager je singleton, takže nepotřebuji žádný self, takže se mi ta akce dá krásně nastavit v initu a ještě closure se zredukuje na jeden řádek, protože nepotřebuji checkovat self. Ale u AccountVC potřebuji v closure sáhnout selfem na username, password a validationErrors.

Reply:
jde to udělat přes `Action(state:)` konstruktor, který jsem ukazoval na cviku 😊 keychainManager je předaný jako parametr v konstruktoru viewModelu, takže na něj `self` nepotřebuješ 😊

Reply:
Oukej, to jsem asi na cvikách nepostřehl 😳 Včera jsem si projížděl commity ke cvikům a narazil jsem na ten state. Můžu to opravit v další branchi? Už jsem tam rozdělal dependency injection a zítra to budu dodělávat, tak abych u toho nemusel zbytečně řešit merge konflitky.

Reply:
Jasny 😊